app/export.go

```diff
@@ -1,185 +1,89 @@
 package app
 
 import (
+	"encoding/json"
+	"time"
+
 	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	tmcrypto "github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/encoding"
+	pc "github.com/tendermint/tendermint/proto/tendermint/crypto"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+
+	"github.com/confio/tgrade/x/poe/contract"
 )
 
 // ExportAppStateAndValidators exports the state of the application for a genesis
 // file.
 func (app *TgradeApp) ExportAppStateAndValidators(
 	forZeroHeight bool, jailAllowedAddrs []string,
 ) (servertypes.ExportedApp, error) {
-	panic("Alex, not implemented")
-	//// as if they could withdraw from the start of the next block
-	//ctx := app.NewContext(true, tmproto.Header{Height: app.LastBlockHeight()})
-	//
-	//// We export at last height + 1, because that's the height at which
-	//// Tendermint will start InitChain.
-	//height := app.LastBlockHeight() + 1
-	//if forZeroHeight {
-	//	height = 0
-	//	app.prepForZeroHeightGenesis(ctx, jailAllowedAddrs)
-	//}
-	//
-	//genState := app.mm.ExportGenesis(ctx, app.appCodec)
-	//appState, err := json.MarshalIndent(genState, "", "  ")
-	//if err != nil {
-	//	return servertypes.ExportedApp{}, err
-	//}
-	//
-	//validators, err := staking.WriteValidators(ctx, app.stakingKeeper)
-	//return servertypes.ExportedApp{
-	//	AppState:        appState,
-	//	Validators:      validators,
-	//	Height:          height,
-	//	ConsensusParams: app.BaseApp.GetConsensusParams(ctx),
-	//}, err
+	if forZeroHeight {
```

Comment:
Can you add a comment (or better make a GitHub issue and link to it) about what would be needed to implement this.

We can do it well after mainnet IMO, but as you are working on it, you can best describe the needed work now.

Reply:
👍  added https://github.com/confio/tgrade/issues/356